### PR TITLE
Support crons with timezone for sidekiq-scheduler

### LINKED ISSUE #2187: Support extracting timezone out of crontab in sidekiq-scheduler
### Issue Description

Hi, just tried out the new auto-crons feature for Sidekiq scheduler. Are there any caveats around it? We have a bunch of running jobs (Rails), but only one shows up in Sentry.

We have a `sidekiq_scheduler.yml.erb`:

<details>

```yaml
process_webhook_requests:
  every: '15s'
  class: StagedWebhookRequestEnqueuerJob

call_status_refresher:
  every: '2h'
  class: CallStatusRefresherEnqueuerJob

delete_webhook_requests:
  cron: '0 21 * * * Europe/Berlin' # 9pm daily
  class: StagedWebhookRequestDeleterJob

process_scheduled_call_requests:
  cron: '*/1 * * * * Europe/Berlin' # Runs every 1 minute
  class: ScheduledCallRequestsProcessorJob
```

</details>

and we have this in our Sidekiq initializer:

<details>

```rb
if ENV.fetch('RUN_SIDEKIQ_CRON').to_bool && Sidekiq.server?
  Sidekiq.configure_server do |config|
    config.on(:startup) do
      scheduler_file_content = File.read(Rails.root.join('config/sidekiq_scheduler.yml.erb'))

      Sidekiq.schedule = YAML.safe_load(
        ERB.new(scheduler_file_content).result
      )

      Sidekiq::Scheduler.reload_schedule!
    end
  end
else
  Sidekiq::Scheduler.enabled = false
end
```

</details>

However, only one job shows up in the UI, after 18+ hours of running:

image

I've checked the logs, and we have the "Injected Sentry Crons monitor checkins into ..." message for all four jobs. And I can also see the transport messages:

image

which makes me wonder if this is a backend issue?

### Reproduction Steps

Add `config.enabled_patches += [:sidekiq_scheduler]`

We did not make any other changes to our individual jobs.

### Expected Behavior

Expected to see all check-ins in the UI

### Actual Behavior

Only saw the one with a schedule `every: 2h`.

### Ruby Version

3.2

### SDK Version

5.14.0

### Integration and Its Version

Rails, Sidekiq, all version 5.14.0

### Sentry Config

```rb

Sentry.init do |config|
  config.enabled_environments = %w[production]
  filterable_params = Rails.application.config.filter_parameters
  filter = ActiveSupport::ParameterFilter.new(filterable_params)
  config.before_send = lambda do |event, _hint|
    filter.filter(event.to_hash)
  end
  config.breadcrumbs_logger = [:active_support_logger]
  config.excluded_exceptions -= ['ActiveRecord::RecordNotFound']
  config.excluded_exceptions += ['Errors::ExpectedError']
  config.enabled_patches += [:sidekiq_scheduler]
end
```

### PR DESCRIPTION
**Pull request by sl0thentr0py**
closes #2187

### CODE REVIEW

**Review by bitsandfoxes — APPROVED** [review]
:shipit:

File: sentry-sidekiq/spec/spec_helper.rb [hunk 1]
```diff
@@ -133,8 +133,10 @@ def perform
   end
 end
 
+class HappyWorkerForCron < HappyWorker; end
```
Comment (st0012) [inline sentry-sidekiq/spec/spec_helper.rb:136]: Why do we need another `HappyWorkerForCron`?

File: sentry-sidekiq/lib/sentry/sidekiq-scheduler/scheduler.rb [hunk 2]
```diff
@@ -37,7 +37,18 @@ def new_job(name, interval_type, config, schedule, options)
         # so we convert it to minutes before passing in to the monitor.
         monitor_config = case interval_type
           when "cron"
-            Sentry::Cron::MonitorConfig.from_crontab(schedule)
+            parsed_cron = ::Fugit.parse_cron(schedule)
```
Comment (st0012) [inline sentry-sidekiq/lib/sentry/sidekiq-scheduler/scheduler.rb:40]: Can we mention the origin of `Fugit` in a comment? It's a second-level dependency of `sidekiq-scheduler`, through the `rufus-scheduler` gem.

File: sentry-sidekiq/spec/spec_helper.rb (hunk 1 above)
Comment (sl0thentr0py) [reply sentry-sidekiq/spec/spec_helper.rb:136]: to keep each class isolated after patching, since we don't have a 'reverse prepend

Comment (st0012) [reply sentry-sidekiq/spec/spec_helper.rb:136]: No I mean we already had this worker before this PR, so now we have 2:

https://github.com/getsentry/sentry-ruby/blob/9b24de9d0a9019d99e2db56d7c5c05d0a8c16e99/sentry-sidekiq/spec/spec_helper.rb#L136-L137

Comment (sl0thentr0py) [reply sentry-sidekiq/spec/spec_helper.rb:136]: ah hmm that was probably copypasta error, thx

File: sentry-sidekiq/lib/sentry/sidekiq-scheduler/scheduler.rb (hunk 2 above)
Comment (sl0thentr0py) [reply sentry-sidekiq/lib/sentry/sidekiq-scheduler/scheduler.rb:40]: https://github.com/getsentry/sentry-ruby/pull/2215

**Pull request merged**